src/control/server/storage/bdev/backend.go

```diff
@@ -302,12 +306,57 @@ func detectVMD() ([]string, error) {
 	return vmdAddrs, nil
 }
 
-// Prepare will execute SPDK setup.sh script to rebind PCI devices as selected
-// by bdev_include and bdev_exclude white and black list filters provided in the
-// server config file. This will make the devices available though SPDK.
+// cleanHugePages removes hugepage files with pathPrefix that are owned by the
+// user with username tgtUsr.
+func cleanHugePages(pathPrefix, tgtUsr string) (int, error) {
+	pattern := pathPrefix + "*"
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		return 0, errors.Wrapf(err, "files matching %q", pattern)
+	}
+
+	toRemove := make([]string, 0, len(matches))
+	for _, path := range matches {
+		username, err := common.GetFileOwner(path)
+		if err != nil {
+			if os.IsNotExist(errors.Cause(err)) {
+				continue
+			}
+			return 0, err
+		}
+		if username != tgtUsr {
+			continue
+		}
+		toRemove = append(toRemove, path)
+	}
+
+	var nrRemoved int
```

Comment:
From an efficiency standpoint, you might consider reimplementing this using https://golang.org/pkg/path/filepath/#Walk to walk a path, skipping files that don't match the pattern. This should also allow you to write more comprehensive test coverage for your test function, as you can feed it synthetic inputs to test different scenarios.